### ISSUE TITLE
Source codes can not access inside China

### ISSUE THREAD

**Issue by jinfagang** [microsoft/onnxruntime#1601]
As somewhat known reason, we are behind the GFC. So that those git url can not access:

```
Cloning into 'third_party/libFuzzer'...
fatal: unable to access 'https://chromium.googlesource.com/chromium/llvm-project/llvm/lib/Fuzzer/': Failed to connect to chromium.googlesource.com port 443: Connection timed out
fatal: clone of 'https://chromium.googlesource.com/chromium/llvm-project/llvm/lib/Fuzzer' into submodule path 'third_party/libFuzzer' failed
Failed to recurse into submodule path 'cmake/external/grpc/third_party/bloaty'
Failed to recurse into submodule path 'cmake/external/grpc'

```

I think it would be help if change those url to accessable url that comes with google, which blocked inside China.

**Comment by snnn** [comment]
@pranavsharma FYI.

**Comment by jywu-msft** [comment]
can we convert some of our submodules to cmake external project?
that would address the issue of always pulling source for code optional components.
git clone of onnxruntime takes forever because of all the nested submodules that were added the past few months. it also slows down CI pipelines.

start by making grpc a cmake external project. looks like there is a submodule in there that cannot be accessed behind GFC.

**Comment by jinfagang** [comment]
@jywu-msft  I agree with that.

**Comment by snnn** [comment]
resolved in #2744
